client/containers/Pub/pubHooks.ts

```diff
@@ -19,3 +22,43 @@ export const usePubData = () => {
 	const { pubData } = useContext(PubContext);
 	return pubData;
 };
+
+const getRenderedValue = (note) => {
+	const hasRenderedStructuredValue = note.renderedStructuredValue?.json?.length;
+	return hasRenderedStructuredValue ? note.renderedStructuredValue.html : note.unstructuredValue;
+};
+
+export const usePubNotes = () => {
+	const {
+		noteManager,
+		collabData: {
+			editorChangeObject: { view },
+		},
+	} = usePubContext();
+	const { citations = [], footnotes = [] } = view ? getNotes(view.state.doc) : {};
+	const [notes, setNotes] = useState(noteManager.notes);
+	useEffect(() => noteManager.subscribe(setNotes), [noteManager]);
+	const isNumberList = noteManager.citationInlineStyleKind === 'count';
+	const renderedFootnotes = footnotes.map((footnote, index) => ({
+		...footnote,
+		...notes[footnote.structuredValue],
+		number: index + 1,
+	}));
+	const renderedCitations = citations
```

Comment:
I'd like to suggest writing this in a way that:

1. Doesn't trigger a `sort()` every render
2. Doesn't do any computation inside of the `sort()` body

The general idea would be:

1. Wrap the whole thing in `useMemo`. I'd expect this hook body to be run hundreds or thousands of times for every time this value _needs_ to be re-rendered (because `citations` or `notes`) changed, so this will really add up.
2. Pre-compute the values that you'll sort by, associated to the values you want to sort by `unstructuredValue`
3. Sort by those values

I remember that we talked about how it would be nice to have a lodash-style utility that lets you sort an array by a key (either a property name or a computed value from each object). If you're so inclined this could be a good place to create and use that utility, since it would accomplish steps 2 and 3.

Here's a sketch of what I have in mind (apologies if this doesn't run!)

```ts
const renderedCitations = useMemo(() => {
    const unsortedRenderedCitations = citations.map((citation, index) => ({
        ...citation,
        ...notes[citation.structuredValue],
        ...(isNumberList && { number: index + 1 }),
    }));

    // You could also use a Map here and hold the rendered citations themselves
    // instead of their structured values.
    const sortableValuesByStructuredValue: Record<string, string> = {};
    unsortedRenderedCitations.forEach(citation => {
        sortableValuesByStructuredValue[citation.structuredValue] = getRenderedValue(citation)
            .replace(/(<([^>]+)>)/gi, '')
            .toLowerCase()
            .trim();
    });

    return unsortedRenderedCitations.sort((a, b) => {
        const preppedA = sortableValuesByStructuredValue[a.structuredValue];
        const preppedB = sortableValuesByStructuredValue[b.structuredValue];
        return preppedA > preppedB ? 1 : -1;
    });
}, [citations, notes]);
```